patches/net/minecraft/world/gen/feature/structure/EndCityStructure.java.patch

```diff
@@ -0,0 +1,67 @@
+--- a/net/minecraft/world/gen/feature/structure/EndCityStructure.java
++++ b/net/minecraft/world/gen/feature/structure/EndCityStructure.java
+@@ -1,7 +1,14 @@
+ package net.minecraft.world.gen.feature.structure;
+ 
++import java.util.List;
+ import java.util.Random;
++
++import carpet.CarpetSettings;
++import com.google.common.collect.Lists;
++import net.minecraft.entity.EntityType;
++import net.minecraft.entity.monster.EntityShulker;
+ import net.minecraft.init.Biomes;
++import net.minecraft.nbt.NBTTagCompound;
+ import net.minecraft.util.Rotation;
+ import net.minecraft.util.SharedSeedRandom;
+ import net.minecraft.util.math.BlockPos;
+@@ -16,6 +23,8 @@
+ 
+ public class EndCityStructure extends Structure<EndCityConfig>
+ {
++    private static final List<Biome.SpawnListEntry> spawnList = Lists.newArrayList(new Biome.SpawnListEntry(EntityType.SHULKER, 10, 4, 4));
++
+     protected ChunkPos getStartPositionForPosition(IChunkGenerator<?> chunkGenerator, Random random, int x, int z, int spacingOffsetsX, int spacingOffsetsZ)
+     {
+         int i = chunkGenerator.getSettings().getEndCityDistance();
+@@ -79,6 +88,11 @@
+         return 9;
+     }
+ 
++    public List<Biome.SpawnListEntry> getSpawnList()
++    {
++        return spawnList;
++    }
++
+     private static int getYPosForStructure(int chunkX, int chunkY, IChunkGenerator<?> generatorIn)
+     {
+         Random random = new Random((long)(chunkX + chunkY * 10387313));
+@@ -140,5 +154,28 @@
+             {
+                 return this.isSizeable;
+             }
++
++            // [CM] Fix EndCity saving
++            @Override
++            public void writeAdditional(NBTTagCompound tagCompound)
++            {
++                super.writeAdditional(tagCompound);
++                if (CarpetSettings.getBool("endCitySavingFix"))
++                {
++                    // Called the tag Valid because it is called Valid in the VillageStructure too
++                    // and it is used to invalidate the Structure
++                    tagCompound.setBoolean("Valid", this.isSizeable);
```

Comment:
I will be frank - not fond of this one. Unless it fixes some critical issue with vanilla, then I would abstain from it, even if it is a valid issue. Especially the comment doesn't convince me - villages are the only temporary structures - the rest is permanent, and extra tags may not be necessary.

Reply:
Well In 1.13 the end city saving is bugged.
Just try this: 
Enter the end and type "/locate EndCity" in the console and teleport there.
Then teleport back to end spawn island. Save and restart the server and try "/locate EndCity" again.
It will not find the End City of the first run any more.
While debugging I noticed that the End City, once generated and saved to file get's invalidated on reload.

But if you like I can remove it.

Reply:
Is it only with the EndCity though?
It maybe considered intended cause most people don't want the same EndCity to appear everytime u run /locate.

Reply:
It is also with Woodland Mansions.
But you might be right, it could be intended.